yarpl/include/yarpl/observable/Observer.h

```diff
@@ -48,6 +48,10 @@ class Observer : public virtual Refcounted {
     return subscription_->isCancelled();
   }
 
+  bool isUnsubscribedOrTerminated() const {
+    return !subscription_ || subscription_->isCancelled();
+  }
```

Comment:
Races with calls to `onComplete/onError` which may reset `subscription_`

Reply:
that is true. This is not thread safe now and we need to address that. See my later comment.